add search functionality

AndrewPomo opened the pull request.

AndrewPomo committed:
add search functionality

albullington reviewed (commented):
Hey @AndrewPomo I like how you're filtering for the reviews which match the search query and then creating a new array with those objects. I think a naming tweak could make it clearer for others to read your code -- toRender could be searchMatched or filteredReviews. You might consider this for ReviewsHolder, HeaderHolder, and StarsHolder too -- ReviewsContainer or maybe dropping Holder altogether would help clarify what each component is for.

AndrewPomo committed:
Clean up search implementation

AndrewPomo commented:
Changed to container for everything. As for `toRender`, I added "reviews" to the front, but since it's not just searched or filtered reviews that fall under that name (the initial render uses `toRender` as well) I'm sticking with the `toRender` language.
